### PR TITLE
Deprecate AutoFinishScopeManager

### PR DESCRIPTION
**Pull request by carlosalberto**
The last fix before releasing 0.32 ;)

@yurishkuro

### CODE REVIEW

File: opentracing-util/src/main/java/io/opentracing/util/AutoFinishScopeManager.java [hunk 1]
```diff
@@ -18,6 +18,13 @@
 
 import java.util.concurrent.atomic.AtomicInteger;
 
+/**
+ * @deprecated use {@link ThreadLocalScopeManager} instead.
```
Comment (sjoerdtalsma) [inline opentracing-util/src/main/java/io/opentracing/util/AutoFinishScopeManager.java]: Does this explain you you can still 'automatically' finish spans when scope is closed?
Or do we not want that anymore?

i.e. Please describe the _reason_ for deprecating this functionality

**Comment by yurishkuro** [comment]
The operating mode this classes proposes is in contrast with the main philosophy of 0.32 where we deprecated autoFinish flag in  activate (), because it too easily leads to programming mistakes of using try-with-resource on the scope and not having the right span to handle errors.

I think we should stick with one recommended instrumentation pattern, shown in 0.32 readme, instead of having the instrumentation be dependent on which scope manager is in use.

**Comment by carlosalberto** [comment]
Also, the historical context is that we wanted to preserve this for people migrating from 0.30. 

That being said, I will add this and @yurishkuro 's note and then merge.

File: opentracing-util/src/main/java/io/opentracing/util/AutoFinishScopeManager.java (hunk 1 above)
Comment (carlosalberto) [reply opentracing-util/src/main/java/io/opentracing/util/AutoFinishScopeManager.java]: Added a (very) small note. Let me know if you think we need further clarification/details ;)

File: opentracing-util/src/main/java/io/opentracing/util/AutoFinishScope.java [hunk 2]
```diff
@@ -19,6 +19,17 @@
 
 import java.util.concurrent.atomic.AtomicInteger;
 
+/**
+ * @deprecated use {@link ThreadLocalScope} instead.
+ * The operation mode of this class contrasts with the 0.32
+ * deprecation of auto finishing {@link Span}s upon {@link Scope#close()}.
```
Comment (yurishkuro) [inline opentracing-util/src/main/java/io/opentracing/util/AutoFinishScope.java]: Could also add a link to this issue for more background.

Comment (sjoerdtalsma) [reply opentracing-util/src/main/java/io/opentracing/util/AutoFinishScope.java]: Great idea!

Comment (carlosalberto) [reply opentracing-util/src/main/java/io/opentracing/util/AutoFinishScope.java]: Done ;)

**Comment by carlosalberto** [comment]
Hey hey guys - if you are ok with this, I will merge later today, and issue the 0.32 release right after ;)

Let me know.

**Pull request merged**